pypowsybl/network.py

```diff
@@ -4124,6 +4124,39 @@ def remove_extensions(self, extension_name: str, ids: _Union[str, _List[str]]) -
             ids = [ids]
         _pp.remove_extensions(self._handle, extension_name, ids)
 
+    def get_connectables_order_positions(self, voltage_level_id: str) -> _DataFrame:
+        """
+        Gets the order positions of every connectable of a given voltage level in a dataframe.
+        """
+        series_array = _pp.get_connectables_order_positions(self._handle, voltage_level_id)
+        return _create_data_frame_from_series_array(series_array).sort_values(by=['order_position'])
+
+    def get_unused_order_positions_before(self, busbar_section_id: str) -> _Optional[pd.Interval]:
+        """
+        Gets all the available positions between the lowest used position of a given busbar section and the
+        highest used position of the busbar section with the section index equal to the section index of the given
+        busbar section minus one. The result is a list with the lowest available value as the first integer and the
```

Comment:
I find it weird to have only those 2 extreme values and not the available values that we could have in-between.
The fact that we use and interval as return type can let the user believe that all values in-between are available.

Actually, I am really not convinced that the 2 methods "get_unused_positions" are useful, because the "get_connectables_order_positions" already provides us all the occupied slots, and therefore also the free slots, or am I missing something ?

Reply:
Actually, these methods do give all the available positions on the "right" or "left" of the busbar section. Everything in the interval is available.
Let's say you have two busbar sections separated by a disconnector. One has feeders at positions 5, 6, 7 and the other has feeders at positions 10,11.
Then get_unused_order_positions_before(bbs1) would return [-infinity, 4]
get_unused_order_positions_after(bbs1) and get_unused_order_positions_before(bbs2) would return [8,9] and finally
get_unused_order_positions_after(bbs2) would return [12, +infinity].
It is quite practical if one wants to automate the computation of the position : one can then just get the available positions depending on whether one wants to put the equipment on the right or on the left and then for example take the value in the middle on the interval. 
The other method get_connectables_order_positions is easier to use to put a feeder between two feeders or with displaying the taken positions. 
Is it that way that you understood it?

Reply:
I will add some documentation, because it is not very clear or easy to explain.